### PR TITLE
Newrelic agent v3

### PR DESCRIPTION
**Pull request by Darkemon**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,22 @@
-module github.com/gettaxi/newrelicutil
+module github.com/gettaxi/newrelicutil/v2
```
Comment (artuross) [inline go.mod]: Maybe let's also change the import path to `gtforge`?

**Review by artuross — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (Darkemon) [reply go.mod]: Good catch, missed that.

Comment (Darkemon) [reply go.mod]: Fixed https://github.com/gtforge/newrelicutil/pull/1/commits/fc74a0537175d0cd02ffdd4f7fa773a48a3871f0

**Review by artuross — APPROVED** [review]

File: newrelicutil.go [hunk 2]
```diff
@@ -5,42 +5,42 @@ import (
...
+func Transaction(ctx context.Context) *newrelic.Transaction {
+	return newrelic.FromContext(ctx)
 }
 
 // WithTransaction puts the New Relic Transaction object to the given context
 // and returns the new context.
-func WithTransaction(ctx context.Context, txn newrelic.Transaction) context.Context {
-	return context.WithValue(ctx, transaction, txn)
+func WithTransaction(ctx context.Context, txn *newrelic.Transaction) context.Context {
+	return newrelic.NewContext(ctx, txn)
```
Comment (shilkin) [inline newrelicutil.go:28]: I would also leave the old logic:
```go
func Transaction(ctx context.Context) newrelic.Transaction {
        if txn := newrelic.FromContext(ctx); txn != nil {
            return txn
        }

        // for potential old users -- we will delete this after migration
	if txn, ok := ctx.Value(transaction).(newrelic.Transaction); ok {
		return txn
	}

	return nil
}

func WithTransaction(ctx context.Context, txn newrelic.Transaction) context.Context {
        ctx := newrelic.NewContext(ctx, txn)

        // for potential old users -- we will delete this after migration
	return context.WithValue(ctx, transaction, txn)
}
```

Comment (Darkemon) [reply newrelicutil.go:28]: I think it doesn’t make sense as the context key isn’t exposed to user and context is populated with the values only by `Transaction` and read by `WithTransaction`.

Comment (shilkin) [reply newrelicutil.go:28]: This is a potential use-case. I made up this. But it's worth checking.
```
service -- uses utils@v2:  newrelicutils.WithTransaction(ctx, tx) sets new context key
|
package -- uses utils@v1: newrelicutils.Transaction(ctx) reads old context key
```

Comment (Darkemon) [reply newrelicutil.go:28]: in any case utils@v1.Transaction will return nil as it won't be able to assert to Transaction interface

Comment (Darkemon) [reply newrelicutil.go:28]: After a discussion, we decided that it doesn't make sense to try to solve compatibility problem between newrelic agent v2 and v3 - it's almost impossible. It's better to give an opportunity to fallback to newrelicutil with agent v2, therefore newrelicutil with the agent v3 will get the tag v2.0.0.

**Review by shilkin — APPROVED** [review]

**Pull request merged**